addons/mail/models/mail_tracking_value.py

```diff
@@ -113,22 +113,63 @@ def _create_tracking_values(self, initial_value, new_value, col_name, col_info,
+                initial_value = (0, '')
+            elif isinstance(initial_value, models.BaseModel):
+                initial_value = (initial_value.id, initial_value.display_name)
+
+            if not new_value:
+                new_value = (0, '')
+            elif isinstance(new_value, models.BaseModel):
+                new_value = (new_value.id, new_value.display_name)
+
             values.update({
-                'old_value_integer': initial_value.id if initial_value else 0,
-                'new_value_integer': new_value.id if new_value else 0,
-                'old_value_char': initial_value.display_name if initial_value else '',
-                'new_value_char': new_value.display_name if new_value else ''
+                'old_value_integer': initial_value[0],
+                'new_value_integer': new_value[0],
+                'old_value_char': initial_value[1],
+                'new_value_char': new_value[1]
             })
-        elif col_info['type'] in {'one2many', 'many2many'}:
+        elif col_info['type'] in {'one2many', 'many2many', 'tags'}:
+            if not initial_value:
+                old_value_char = ''
+            elif isinstance(initial_value, models.BaseModel):
+                old_value_char = ', '.join(initial_value.mapped('display_name'))
+            else:
+                old_value_char = ', '.join(value[1] for value in initial_value)
+            if not new_value:
+                new_value_char = ''
+            elif isinstance(new_value, models.BaseModel):
+                new_value_char = ', '.join(new_value.mapped('display_name'))
+            else:
+                new_value_char = ', '.join(value[1] for value in new_value)
+
             values.update({
-                'old_value_char': ', '.join(initial_value.mapped('display_name')) if initial_value else '',
-                'new_value_char': ', '.join(new_value.mapped('display_name')) if new_value else '',
+                'old_value_char': old_value_char,
+                'new_value_char': new_value_char,
             })
         else:
             raise NotImplementedError(f'Unsupported tracking on field {field.name} (type {col_info["type"]}')
 
         return values
 
+    @api.model
+    def _create_tracking_values_property(self, initial_value, col_name, col_info, record):
+        """Generate the values for the <mail.tracking.values> corresponding to a property."""
+        col_info = col_info | {'type': initial_value['type'], 'selection': initial_value.get('selection')}
+
+        field_info = {
+            'desc': f"{col_info['string']}: {initial_value['string']}",
+            'name': col_name,
+            'type': initial_value['type'],
+        }
+        value = initial_value.get('value', False)
+        if value and initial_value['type'] == 'tags':
+            value = [t for t in initial_value.get('tags', []) if t[0] in value]
+
+        values = self.env['mail.tracking.value']._create_tracking_values(
```

Comment:
A bit too much valuye, values, ... :sweat_smile: 
```suggestion
        tracking_values = self.env['mail.tracking.value']._create_tracking_values(
```